import-extension/src/test/kotlin/com/pkware/detekt/extensions/rules/staticImport/EnforceStaticImportTest.kt

```diff
@@ -0,0 +1,309 @@
+package com.pkware.detekt.extensions.rules.staticImport
+
+import com.google.common.truth.Truth
+import io.github.detekt.test.utils.KotlinCoreEnvironmentWrapper
+import io.github.detekt.test.utils.createEnvironment
+import io.gitlab.arturbosch.detekt.api.SourceLocation
+import io.gitlab.arturbosch.detekt.test.TestConfig
+import io.gitlab.arturbosch.detekt.test.assertThat
+import io.gitlab.arturbosch.detekt.test.compileAndLintWithContext
+import org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.provider.Arguments
+import java.io.File
+
+private const val METHODS = "methods"
+
+class EnforceStaticImportTest {
+    private lateinit var wrapper: KotlinCoreEnvironmentWrapper
+    private lateinit var env: KotlinCoreEnvironment
+
+    // We need to set up a custom KotlinCoreEnvironment and add the Truth and Arguments class paths to that environment,
+    // so they are available to generate the BindingContext for detekt rules under test with the
+    // EnforceStaticImportExtension.
+    @BeforeEach
+    fun setUp() {
+        wrapper = createEnvironment(
+            additionalRootPaths = listOf(
+                File(Truth::class.java.protectionDomain.codeSource.location.path),
+                File(Arguments::class.java.protectionDomain.codeSource.location.path)
+            )
+        )
+        env = wrapper.env
+    }
+
+    @AfterEach
+    fun tearDown() {
+        wrapper.dispose()
+    }
+
+    @Test
+    fun `rule report based on custom configuration`() {
+
+        val code = """
+            fun main() {
+                val value = 3.3
+                value = Math.floor(value)
+                print("3")
+            }
+            """
+
+        val findings = EnforceStaticImport(
+            TestConfig(mapOf(METHODS to listOf("java.lang.Math.floor", "kotlin.io.print")))
+        ).compileAndLintWithContext(env, code)
+
+        assertThat(findings).hasSize(1)
```

Comment:
`hasSourceLocations` checks for size and the message format is more valuable, as it will fail with the contents in the message.
```suggestion
```

Reply:
so detekts own tests still use both the size and the has sourceLocations separately as the size will fail fast with that specific error:

see ~detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCallSpec.kt in https://github.com/detekt/detekt/tree/v1.20.0/detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style

I will clean up the uses where the size = 1 but leave the check in place where the size > 1

Reply:
Actually, don't. I just realized I was seeing the message and tests that were using `assertSize`. Keep them! I've made a huge mistake!